src/streamlink/plugin/plugin.py

```diff
@@ -398,6 +398,15 @@ def get_streams(self, *args, **kwargs):
     def _get_streams(self):
         raise NotImplementedError
 
+    def _get_title(self):
```

Comment:
I think these should be public method (ie. not prefixed with `_`) or possibly `@property`s - this is adding something to the API that other plugin devs might have an opinion on (paging dr. @back-to).

Reply:
plugin developers who are more familiar with the streamlink API than I am could implement `@property`s if it's not too big of a job. Personally I don't know what a `@property` is. Just removed the leading underscore for now.

Reply:
It was more that for other Plugins to give a Title, Author, etc. they would need to implement these methods, so once they have names that's it :) FYI `@propery`: https://docs.python.org/3/library/functions.html#property

Reply:
I've gone into deep meditation on `@property`, and decided that I don't understand the benefit of creating this structure versus just defining methods as they're written. Seems like an appropriate place to use it, but I just don't understand what benefit it has.

Not sure what "once they have names that's it" means, sorry :s

Reply:
For @property I just think it looks nicer to write `plugin.title` vs. `plugin.get_title()` - but probably `get_title` is better ;)

Reply:
I mean, I want to make whatever changes are necessary to get the feature into streamlink. But personally I like `get_title()` better because it's clear that it's explicitly calling a function. The reader would have to know that `plugin.title` is this `@property` structure, or else while skimming, they could just think it's a static value that doesn't do anything.

If not using `@property` makes it harder to refactor later and you want it to be a `@property` then I'll make it a property. I just want people to care about the feature and work with me.

Reply:
Yeah, `get_title()` is better as it might have to make extra calls, like in your YouTube Twitch examples.

Reply:
Ok, thanks... :-)